cedar-policy/src/api.rs

```diff
@@ -3387,30 +3387,50 @@ impl Context {
 
     /// Combine two [`Context`]s, returning an error if they contain overlapping keys
     pub fn merge(self, context: Self) -> Result<Self, ContextCreationError> {
-        Self::from_pairs(self.values().into_iter().chain(context.values()))
+        Self::from_pairs(self.into_iter().chain(context))
     }
+}
 
-    /// Consume the context and return a `HashMap` of its elements. Note that
-    /// context elements are evaluated when the context is constructed, so the
-    /// stored values may not exactly match the expressions used as input to
-    /// constructors like [`Context::from_pairs`] or [`Context::from_json_str`].
-    pub fn values(self) -> HashMap<String, RestrictedExpression> {
-        self.0
-            .values()
-            .map(|(k, v)| {
+/// Utilities for implementing `IntoIterator` for `Context`
+mod context {
+    use super::{ast, RestrictedExpression};
+
+    /// `IntoIter` iterator for `Context`
+    #[derive(Debug)]
+    pub struct IntoIter {
+        pub(super) inner: <ast::Context as IntoIterator>::IntoIter,
+    }
+
+    impl Iterator for IntoIter {
+        type Item = (String, RestrictedExpression);
+
+        fn next(&mut self) -> Option<Self::Item> {
+            self.inner.next().map(|(k, v)| {
                 (
                     k.to_string(),
                     match v {
-                        ast::PartialValue::Value(val) => RestrictedExpression(
-                            ast::RestrictedExpr::new_unchecked(ast::Expr::from(val)),
-                        ),
+                        ast::PartialValue::Value(val) => {
+                            RestrictedExpression(ast::RestrictedExpr::from(val))
+                        }
                         ast::PartialValue::Residual(exp) => {
                             RestrictedExpression(ast::RestrictedExpr::new_unchecked(exp))
```

Comment:
maybe a short comment justifying the `_unchecked`, like you have elsewhere in this PR?